### ISSUE TITLE
Exception drawing ThingOverlay

### ISSUE THREAD

**Issue by Suniver** [zymex22/Project-RimFactory-Revived#360]
I have not found another issue with this same bug, but if it's a duplicate and I missed it, sorry in advance.

# Describe the bug
The "Underground belt entrance" is throwing a OutOfRangeException even without items being pulled onto it.
Even if there is an error, it's seems to work properly.

**To Reproduce**
Reproducibility: ?
1. Build an Item Puller (I'm using MkI actually don't know if relevant).
2. Build an Underground belt entrance.
3. Build underground conveyors.
4. Build underground belt exit.
5. Build item angle puller.

**Expected behavior**
Well I expect that there is nor error in the logs, but I looks like it works in game.

**Actual Behavior**
Is throwing an red error in the logs:
```
Exception drawing ThingOverlay for PRF_Conveyor_Connector_To_Floor2198026: System.ArgumentOutOfRangeException
![prf_underground_belt_entrance_exception](https://user-images.githubusercontent.com/77306553/104255515-8ab7a800-5479-11eb-8455-4c704ac72166.png)
: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) [0x00029] in <567df3e0919241ba98db88bec4c6696f>:0 
  at System.ThrowHelper.ThrowArgumentOutOfRangeException () [0x00000] in <567df3e0919241ba98db88bec4c6696f>:0 
  at ProjectRimFactory.AutoMachineTool.Building_BeltConveyor.CarryingThing () [0x00041] in <084c448145cc4528ba86c8c6026b5123>:0 
  at ProjectRimFactory.AutoMachineTool.Building_BeltConveyor.DrawGUIOverlay () [0x00091] in <084c448145cc4528ba86c8c6026b5123>:0 
  at ProjectRimFactory.AutoMachineTool.Building_BeltConveyorUGConnector.DrawGUIOverlay () [0x00001] in <084c448145cc4528ba86c8c6026b5123>:0 
  at (wrapper dynamic-method) Verse.ThingOverlays.Verse.ThingOverlays.ThingOverlaysOnGUI_Patch1(Verse.ThingOverlays)
Verse.Log:Verse.Log.Error_Patch2(String, Boolean)
Verse.ThingOverlays:Verse.ThingOverlays.ThingOverlaysOnGUI_Patch1(ThingOverlays)
RimWorld.MapInterface:RimWorld.MapInterface.MapInterfaceOnGUI_BeforeMainTabs_Patch3(MapInterface)
RimWorld.UIRoot_Play:RimWorld.UIRoot_Play.UIRootOnGUI_Patch1(UIRoot_Play)
Verse.Root:Verse.Root.OnGUI_Patch1(Root)
```

**Screenshots**
![prf_underground_belt_entrance_exception](https://user-images.githubusercontent.com/77306553/104255614-c0f52780-5479-11eb-941f-f7753edc1524.png)

**Rimworld (please complete the following information):**
 - RimWorld Version: 1.2.2753
 - PRF Version: 2.5.2
 - Link to gist bug report: https://gist.github.com/HugsLibRecordKeeper/c4a31c04c1b2a216e721551d9b84270f
 - Type: Exception
 - Severity: Not game breaking, but red log.
 

**Modlist (please complete the following information):**
 - Does this problem exist if only prf is installed: Yes
 - Your modlist (if needed): In my gist link.
 - Do you know what mod conflicts?: None.

**Additional context**
I found out because I had another issue I don't know related or not to this one but unable to reproduce: I am pulling out of a big JDS - Simple Storage deep storage full of stone chunks, also using stackable chunks. My belt is pulling into a single tile vanilla stockpile, and the belt (or the angle puller ?) destroyed all my chunks on the belts... whend the destination stockpile is full. After this the belts seemed to be freezed with a stauts "Placing. Pending placement [0]" or something. Was unable to destroy item via game, had to destroy with devmod, will try to reproduce.

**Comment by lilwhitemouse** [comment]
Are you using the most recent Steam version of PRF?

**Comment by Suniver** [comment]
I'm using the Mod Manager by Fluffy, not a local version, should be updated auto from steam. In the mod manager I see 2.5.2, How can I see a more specific version number ?

**Comment by lilwhitemouse** [comment]
Nope, that's the most recent version out there!

**Comment by lilwhitemouse** [comment]
Sniper offers this up as potentially related:

https://gist.github.com/HugsLibRecordKeeper/c4a31c04c1b2a216e721551d9b84270f#file-output_log-txt-L5009

**Comment by Suniver** [comment]
So I made so more test to try to help you troubleshoot the problem.

I created a new game with only the following mods:
- Harmony
- Core
- Royalty
- Hugslib
- PRF

This time a created several setups:

1. Puller Mk1 => Conveyor blets (level1) => Auto-assembler Mk1. 
No errors in the console log.
![prf_simple_belt_setup_no_error](https://user-images.githubusercontent.com/77306553/104450109-e2e8ca00-559f-11eb-8d53-44a59675f528.png)

2. Puller Mk1 => Conveyor blets (level1) => Puller Mk1 => Auto-assembler Mk1
Exception in the console log. The gist: https://gist.github.com/f53892039c4cc933167a80de3d2b50cc
![prf_simple_belt_into_puller_setup_error](https://user-images.githubusercontent.com/77306553/104451524-f7c65d00-55a1-11eb-9a35-9973d6ac55fb.png)


3. Puller Mk1 => Conveyor belts (level 1) => Puller Mk1 => Small Steel Crate => Auto-assembler Mk1.
The exception is in the console log. Here is the gist: https://gist.github.com/HugsLibRecordKeeper/d3be18664395a6b9a81e9922a93ac62c
![prf_simple_belt_and_crate_setup_error](https://user-images.githubusercontent.com/77306553/104450716-d022c500-55a0-11eb-9f43-654d95b637e8.png)

I made a setup with faster belts and dame the error is here when using a puller at the end of the line before the auto-assembler. Was thinking the problem is with the puller at the end but:

4. Puller Mk1 => Underground belt entrance => Underground belts => Underground belt exit => Auto-assembler Mk1.
We still have the exception here. Gist: https://gist.github.com/6fee15dd17cbb82fa0c59d8af27b6ad3
![prf_undeground_belt_setup_error](https://user-images.githubusercontent.com/77306553/104452427-488a8580-55a3-11eb-950b-3f8aadd05951.png)

5. Puller Mk1 => Underground belt entrance => Underground belts => Underground belt exit => Angle Puller => Auto-assembler Mk1.
Exception here, but things a getting worse... chunks are disappearing into the void from the belts. Not sure this is the same problem, so I will write a new issue with more details. But here is the gist: https://gist.github.com/663d46ac969e8e690a6e50deab37d679
![prf_undeground_belt_angle_puller_setup_error_chunks_destroyed](https://user-images.githubusercontent.com/77306553/104453434-b1263200-55a4-11eb-8f94-f62b9e0f7993.png)

If you need me to make more test or more details feel free to ask me!

**Comment by lilwhitemouse** [comment]
Apparently GitHub does not differentiate between "closes" and "hopefully closes" - well, we can reopen this if we need to.

**Comment by lilwhitemouse** [comment]
Okay, that fix will only fix the first part of this, which does not affect the 2nd half

**Comment by Suniver** [comment]
I give more details on the 2nd part in #362

**Comment by lilwhitemouse** [comment]
I think I have reproduced the error; I think it's something to do with how the SAL robot takes items directly from conveyor belts.

**Comment by lilwhitemouse** [comment]
Closing this with remaining part of bug in #362

**Comment by aef8234** [comment]
This same exact issue is happening to me too so it might not have been fixed.
Here's my log: https://gist.github.com/7dcb16709110615a90d472cde3b584e3

I saw a pattern though shorter belts and belts underneath roofing like sensor panels from expanded roofing will get turned into whatever "placement 0-3" is. Also it isn't just SAL robots, drillers and just all around putting things in a conveyor belt is doing it. Maybe it's lag-related?
